<?xml version="1.0" ?>
<!-- Launch file used to launch the hardware interface (which publish the joint state), with drivers, controllers, robot_description, and robot_state -->
<launch xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:noNamespaceSchemaLocation="https://gist.githubusercontent.com/nalt/dfa2abc9d2e3ae4feb82ca5608090387/raw/roslaunch.xsd">

    <arg name="execute_on_niryo_robot_raspberry_pi_image" default="true" />
    <arg name="simu_gripper" default="true"/>
    <arg name="simu_conveyor" default="true"/>
    <arg name="gazebo" default="false"/>
    <arg name="can_enabled" default="true"/>
    <arg name="ttl_enabled" default="true"/>
    <arg name="hardware_version" default="ned"/>
    <arg name="simulation_mode" default="true"/>
    <arg name="debug" default="false"/>
    <arg name="image_version" default="" />

    <group ns="niryo_robot">
        <group ns="info">
            <param name="image_version" type="string" value="$(arg image_version)" />
            <param name="ros_version" type="string" command="rosversion niryo_robot_bringup" />
        </group>
    </group>

    <!-- Launch controllers -->
    <rosparam file="$(find niryo_moveit_config_standalone)/config/$(arg hardware_version)/ros_controllers.yaml" command="load"/>
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="log"
            args="joint_state_controller niryo_robot_follow_joint_trajectory_controller"/>

    <!-- Load URDF -->
    <arg name="urdf_path" value="$(find niryo_robot_description)/urdf/$(arg hardware_version)/niryo_$(arg hardware_version).urdf.xacro"/>
    <param name="robot_description" command="$(find xacro)/xacro  $(arg urdf_path)"/>

    <!-- Launch Robot state publisher -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen" />
    <!-- <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher"/> -->

    <!--use base config -->
    <include file="$(find niryo_robot_hardware_interface)/launch/niryo_robot_hardware_interface_base.launch.xml" >
        <arg name="hardware_version" value="$(arg hardware_version)" />
        <arg name="simulation_mode" value="$(arg simulation_mode)"/>
        <arg name="simu_gripper" value="$(arg simu_gripper)"/>
        <arg name="simu_conveyor" value="$(arg simu_conveyor)"/>
        <arg name="can_enabled" value="$(arg can_enabled)" />
        <arg name="ttl_enabled" value="$(arg ttl_enabled)" />
    </include>

    <!-- Launch hardware interface  -->
    <node pkg="niryo_robot_hardware_interface" type="niryo_robot_hardware_interface_node"
          name="niryo_robot_hardware_interface" output="screen">
        <param name="hardware_version" value="$(arg hardware_version)"/>
        <param name="gazebo" type="bool" value="$(arg gazebo)"/>
        <param name="can_enabled" type="bool" value="$(arg can_enabled)"/>
        <param name="ttl_enabled" type="bool" value="$(arg ttl_enabled)"/>
        <param name="debug" type="bool" value="$(arg debug)"/>

        <!-- Rosparam definition  -->
        <rosparam file="$(find niryo_robot_hardware_interface)/config/default.yaml" command="load"/>
    </node>

</launch>
